packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -288,3 +288,21 @@ class NoOpDevtoolsHandler implements ResidentDevtoolsHandler {
     return;
   }
 }
+
+/// Convert a [URI] with query parameters into a display format instead
+/// of the default URI encoding.
+@visibleForTesting
+String urlToDisplayString(Uri uri) {
+  final StringBuffer base = StringBuffer(uri.replace(
+    queryParameters: <String, String>{},
+  ).toString());
+  String fencepost = '';
+  for (final MapEntry<String, String> entry in uri.queryParameters.entries) {
+    base.write(fencepost);
```

Comment:
Interesting; I haven't seen strings joined this way before. You could achieve the same effect with `params.keys.map((key) => '$key=${params[key]}').join('&')` ?

Reply:
Yeah that seems much simpler 😅